### PR TITLE
`azurerm_api_management_policy` - fixes an error caused by a migration

### LINKED ISSUE #23015: Microsoft.ApiManagement Error: unexpected segment "policies/policy"
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.4.0

### AzureRM Provider Version

3.70.0

### Affected Resource(s)/Data Source(s)

azurerm_api_management_policy

### Terraform Configuration Files

```hcl
# global-policy.tf
resource "azurerm_api_management_policy" "this" {
  count             = var.global_policy_configuration != "" ? 1 : 0
  api_management_id = data.azurerm_api_management.this.id
  xml_content       = var.global_policy_configuration
}
```

### Debug Output/Panic Output

```shell
Global Policy applied successfully with Azurerm v3.69.0

module.api-global-policy[0].azurerm_api_management_policy.this[0]: Modifying... [id=/subscriptions/<hidden>/resourceGroups/rg-ag-qa01-api-s-01/providers/Microsoft.ApiManagement/service/apim-ag-qa01-api-s-01/policies/policy]
module.api-global-policy[0].azurerm_api_management_policy.this[0]: Modifications complete after 7s [id=/subscriptions/<hidden>/resourceGroups/rg-ag-qa01-api-s-01/providers/Microsoft.ApiManagement/service/apim-ag-qa01-api-s-01/policies/policy]

Upgrading to the latest Azurerm v3.70.0 and now terraform plan fails

Planning failed. Terraform encountered an error while generating this plan.
╷
│ Error: parsing "/subscriptions/<hidden>/resourceGroups/rg-ag-qa01-api-s-01/providers/Microsoft.ApiManagement/service/apim-ag-qa01-api-s-01/policies/policy": unexpected segment "policies/policy" present at the end of the URI (input "/subscriptions/<hidden>/resourceGroups/rg-ag-qa01-api-s-01/providers/Microsoft.ApiManagement/service/apim-ag-qa01-api-s-01/policies/policy")

Fixing the provider version to Azurerm v3.69.0 and plan works again which implies that there is some type of mishap in the Azurerm v3.70.0 affecting the resourse azurerm_api_management_policy.
```

### Expected Behaviour

Plan expected to finish cleanly, like...

```
module.api-global-policy[0].azurerm_api_management_policy.this[0]: Refreshing state... [id=/subscriptions/<hidden>/resourceGroups/rg-ag-qa01-api-s-01/providers/Microsoft.ApiManagement/service/apim-ag-qa01-api-s-01/policies/policy]
No changes. Your infrastructure matches the configuration.
Terraform has compared your real infrastructure against your configuration
and found no differences, so no changes are needed.
```

### Actual Behaviour

Plan fails with error 
```
Planning failed. Terraform encountered an error while generating this plan.
╷
│ Error: parsing "/subscriptions/<hidden>/resourceGroups/rg-ag-qa01-api-s-01/providers/Microsoft.ApiManagement/service/apim-ag-qa01-api-s-01/policies/policy": unexpected segment "policies/policy" present at the end of the URI (input "/subscriptions/<hidden>/resourceGroups/rg-ag-qa01-api-s-01/providers/Microsoft.ApiManagement/service/apim-ag-qa01-api-s-01/policies/policy")
```

### Steps to Reproduce

1. `terraform plan` using Azurerm v3.69.0
2. `terraform init -upgrade` upgrading to Azurerm v3.70.0
3. `terraform plan` now fails

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by mbfrahry**
Fixes #23015

### CODE REVIEW

**Review by jackofallops — APPROVED** [review]
Thanks @mbfrahry - apart from the `goimports` issue LGTM 👍

File: internal/services/apimanagement/migration/policy.go [hunk 1]
```diff
@@ -20,11 +21,11 @@ type ApiManagementPolicyV0ToV1 struct{}
 
 func (ApiManagementPolicyV0ToV1) UpgradeFunc() pluginsdk.StateUpgraderFunc {
 	return func(ctx context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
-		apiMgmtId, err := policy.ParseServiceID(rawState["id"].(string))
+		apiMgmtId, err := parse.ApiManagementID(rawState["id"].(string))
```
Comment (stephybun) [inline internal/services/apimanagement/migration/policy.go:24]: I think these two IDs are identical and we don't need to revert this back to using the older generated parsers which imo we should be removing after doing sdk swaps.

**Review by stephybun — APPROVED** [review]
Thanks @mbfrahry one minor comment but otherwise LGTM 🍶

File: internal/services/apimanagement/migration/policy.go (hunk 1 above)
Comment (mbfrahry) [reply internal/services/apimanagement/migration/policy.go:24]: You are right! I'll change that up now. Thank you thank you

**Pull request merged**